src/main/java/com/google/sps/servlets/EventsServlet.java

```diff
@@ -0,0 +1,119 @@
+package com.google.sps.servlets;
+
+import static com.google.sps.utils.ServletUtils.getParameter;
+import static com.google.sps.utils.StringConstants.*;
+
+import com.google.api.client.util.DateTime;
+import com.google.api.services.calendar.Calendar;
+import com.google.api.services.calendar.model.Event;
+import com.google.api.services.calendar.model.Events;
+import com.google.api.services.calendar.model.EventDateTime;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Strings;
+import com.google.sps.utils.ServletUtils;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Servlet for obtaining the Calendar ID of a group and adding or obtaining events
+ */
+@WebServlet("/events")
+public class EventsServlet extends AbstractEventsServlet {
+  private final int MAX_EVENTS = 50;
+
+  /**
+   * Obtain a list of events from the specified group if given the Calendar ID.
+   * This value should be provided in the query string.
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String calendarId = getParameter(request, GROUP_CALENDARID_PROPERTY);
+
+    if (Strings.isNullOrEmpty(calendarId)) {
+      ServletUtils.printBadRequestError(response, EVENTS_GET_BAD_REQUEST_MESSAGE);
+      return;
+    }
+
+    try {
+      Events events = getEventsList(calendarId); 
+
+      response.setContentType(CONTENT_TYPE_JSON);
+      response.getWriter().println(events.toString());
+    } catch (Exception authenticationError) {
+      response.getWriter().println(EVENTS_GET_INVALID_CALENDARID_MESSAGE);
+    }
+  }
+
+  /**
+   * Create an event and add it to the calendar of the specified group. Query string should contain groupId
+   * POST body should contain: title,  description (opt), location (opt), start time, and end time.
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String groupId = getParameter(request, GROUP_ID_PROPERTY);
+    String eventTitle = getParameter(request, EVENT_TITLE_PROPERTY);
+    String eventStart = getParameter(request, EVENT_START_PROPERTY);
+    String eventEnd = getParameter(request, EVENT_END_PROPERTY);
+
+    if (!Strings.isNullOrEmpty(groupId) && !Strings.isNullOrEmpty(eventTitle) 
```

Comment:
Do the early return here too.

It'll especially help with readability here since the user will immediately be able to tell that you're just checking for bad inputs.